lbrynet/lbrynet_daemon/auth/server.py

```diff
@@ -189,16 +190,39 @@ def _render_message(self, request, message):
         request.finish()
 
     def _check_headers(self, request):
+        return self._check_origin(request) and self._check_referer(request)
+
+    def _check_origin(self, request):
         origin = request.getHeader("Origin")
-        referer = request.getHeader("Referer")
-        if origin not in [None, settings.ORIGIN]:
-            log.warning("Attempted api call from %s", origin)
+        if not self._check_source_of_request(origin):
+            log.warning("Attempted api call from invalid origin: %s", origin)
             return False
-        if referer is not None and not referer.startswith(settings.REFERER):
```

Comment:
okay.  I'll take a closer look at some of the common referrers and fix this up.

BTW - this is a good example of where tests are useful to prevent a regression.